go/cmd/vtcombo/tablet_map.go

```diff
@@ -496,48 +496,9 @@ func (itc *internalTabletConn) SplitQuery(
 	return splits, nil
 }
 
-type streamHealthReader struct {
-	c   <-chan *querypb.StreamHealthResponse
-	err *error
-}
-
-// Recv implements tabletconn.StreamHealthReader.
-// It returns one response from the chan.
-func (r *streamHealthReader) Recv() (*querypb.StreamHealthResponse, error) {
-	resp, ok := <-r.c
-	if !ok {
-		return nil, *r.err
-	}
-	return resp, nil
-}
-
 // StreamHealth is part of tabletconn.TabletConn
-func (itc *internalTabletConn) StreamHealth(ctx context.Context) (tabletconn.StreamHealthReader, error) {
-	result := make(chan *querypb.StreamHealthResponse, 10)
-
-	id, err := itc.tablet.qsc.QueryService().StreamHealthRegister(result)
-	if err != nil {
-		return nil, err
-	}
-
-	var finalErr error
-	go func() {
-		select {
-		case <-ctx.Done():
-		}
-
-		// We populate finalErr before closing the channel.
-		// The consumer first waits on the channel closure,
-		// then read finalErr
-		finalErr = itc.tablet.qsc.QueryService().StreamHealthUnregister(id)
-		finalErr = tabletconn.TabletErrorFromGRPC(vterrors.ToGRPCError(finalErr))
-		close(result)
-	}()
-
-	return &streamHealthReader{
-		c:   result,
-		err: &finalErr,
-	}, nil
+func (itc *internalTabletConn) StreamHealth(ctx context.Context, sendReply func(*querypb.StreamHealthResponse) error) error {
+	return itc.tablet.qsc.QueryService().StreamHealth(ctx, sendReply)
```

Comment:
I think you still need this one here:
err := itc.tablet.qsc.QueryService().StreamHealth(ctx, sendReply)
return tabletconn.TabletErrorFromGRPC(vterrors.ToGRPCError(err))

(code needs to be checked for nil and stuff).

That way we still simulate the gRPC error conversion, going through canonical codes.